Bencodex/IBencodable.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Diagnostics.Contracts;
+using Bencodex.Types;
+
+namespace Bencodex
+{
+    /// <summary>
+    /// Defines a generic <see langword="interface"/> for an object that can be
+    /// encoded into an <see cref="IValue"/>.  Any <see langword="class"/> implementing
+    /// this <see langword="interface"/> should also implement either a constructor
+    /// or a <see langword="static"/> factory method with an <see cref="IValue"/> as a parameter
+    /// for decoding.
+    /// </summary>
+    /// <remarks>
+    /// Note that encoding and decoding mentioned here are different from
+    /// <see cref="Codec.Encode(IValue)"/> and <see cref="Codec.Decode(byte[])"/>.
+    /// </remarks>
+    /// <example>
+    /// The following example shows an implementation of an integer point <see langword="class"/>
+    /// with two distinct methods of decoding, via a constructor and a <see langword="static"/>
+    /// factory method, for illustration:
+    /// <code><![CDATA[
+    /// public class Point : IBencodable
+    /// {
+    ///     public Point(int x, int y)
+    ///     {
+    ///         X = x;
+    ///         Y = y;
+    ///     }
+    ///
+    ///     public Point(IValue bencoded)
+    ///         : this((List)bencoded)
+    ///     {
+    ///     }
+    ///
+    ///     private Point(List bencoded)
+    ///         : this((Integer)bencoded[0], (Integer)bencoded[1])
+    ///     {
+    ///     }
+    ///
+    ///     public int X { get; }
+    ///
+    ///     public int Y { get; }
+    ///
+    ///     public static Point Decode(IValue bencoded)
```

Comment:
It is just a comment.

Since C# 11, the `interface` can have `static abstract` methods but it doesn't support some type like the `Self` type in Rust language. 😞